core/src/main/java/org/apache/iceberg/BaseTable.java

```diff
@@ -63,6 +63,10 @@ public String name() {
     return name;
   }
 
+  public MetricsReporter reporter() {
```

Comment:
What I see in RESTSessionCatalog is that the MetricsReporter itself is not exposed from the class. Rather a MetricsReport is exposed through a function that is given to the BaseTable during creation:
https://github.com/apache/iceberg/blob/6697129a314d98cb793601495f7ebb2ae000b40a/core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java#L315

Reply:
removing this function.